### ISSUE TITLE
roachtest: disk stalls on GCP / Azure lead to somewhat frequent test failures

### ISSUE THREAD

**Issue by renatolabs** [cockroachdb/cockroach#97968]
We see tests fail because of disk stalls with some regularity, and there's a sense that they happen more than is acceptable. We have a support ticket with GCP [1]. According to their support team, we are expected to see improvements in this area 4-6 weeks after Feb 16 -- in other words, we expect these issues to not come up nearly as often by early April at the latest.

This issue is for us to keep track of disk stall failures (by mentioning this issue on failures caused by disk stalls) and to make sure we check on the progress of the fix and close it when we think it's resolved.

[1] <https://console.cloud.google.com/support/cases/detail/v2/42856817?project=cockroach-ephemeral>

Jira issue: CRDB-24992

**Comment by andrewbaptist** [comment]
@renatolabs - I wanted to check what was the next step on this. It appears to still be happening, but the case on the GCP side is closed.

**Comment by srosenberg** [comment]
> @renatolabs - I wanted to check what was the next step on this. It appears to still be happening, but the case on the GCP side is closed.

We can ping GCP support again, but I have little faith in that. I think our chances are higher if we were to switch to `pd-ssd`. Other than cost, which isn't significant, do we still need local ssd?

**Comment by erikgrinaker** [comment]
I'm starting to get really suspicious of these disk stalls. We keep seeing them much more often than I'd expect. How confident are we that this is an infrastructure problem, and not a CRDB/Pebble problem?

cc @jbowens

**Comment by jbowens** [comment]
Agreed, we've been suspicious of them too. We've given it additional looks several times now, and there always appears to be a goroutine stuck in an I/O syscall for a long duration. I'll add some additional observability here.

Interestingly, in that most recent failure in #108680, there are goroutines in I/O syscalls at the time of the `Fatalf`, but none of them match the operation that triggered the `Fatalf`: a `Sync` on the cockroach data directory.

**Comment by stevendanna** [comment]
On DR, in addition to these disk stalls, we've also been seeing tests fail with "exit code 8", which we believe means that we hit some error but then timed out trying to even write the error to the log file...

https://github.com/cockroachdb/cockroach/issues/109233

The assumption has been infra/test-runner-overload.  But, I agree it is suspicious this is all happening now.

**Comment by msbutler** [comment]
Had another instance of this https://github.com/cockroachdb/cockroach/issues/113823

**Comment by nicktrav** [comment]
There's not much action to take here.

WAL Failover was developed to help mitigate this kind of failure mode.
